src/drivers/mount/mount.cpp

```diff
@@ -0,0 +1,467 @@
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file mount.cpp
+ * @author Leon Müller (thedevleon)
+ * MAV_MOUNT driver for controlling mavlink gimbals, rc gimbals/servors and
+ * future kinds of mounts.
+ *
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include <string.h>
+#include <sys/types.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <systemlib/err.h>
+#include <systemlib/systemlib.h>
+
+#include "mount_mavlink.h"
+#include "mount_rc.h"
+#include "mount_onboard.h"
+
+#include <uORB/uORB.h>
+#include <uORB/topics/vehicle_roi.h>
+#include <uORB/topics/vehicle_global_position.h>
+#include <uORB/topics/vehicle_mount.h>
+#include <uORB/topics/position_setpoint_triplet.h>
+#include <uORB/topics/manual_control_setpoint.h>
+
+#include <px4_config.h>
+
+/* thread state */
+static volatile bool thread_should_exit = false;
+static volatile bool thread_running = false;
+static int mount_task;
+typedef enum { IDLE, MAVLINK, RC, ONBOARD } mount_state_t;
+static mount_state_t mount_state = IDLE;
+
+/* functions */
+static void usage(void);
+static void mount_update_topics(void);
+static bool get_params(void);
+static int mount_thread_main(int argc, char *argv[]);
+__EXPORT int mount_main(int argc, char *argv[]);
+
+/* uORB subscriptions */
+static int vehicle_roi_sub = -1;
+static int vehicle_global_position_sub = -1;
+static int vehicle_mount_sub = -1;
+static int position_setpoint_triplet_sub = -1;
+static int manual_control_setpoint_sub = -1;
+
+static struct vehicle_roi_s *vehicle_roi;
+static bool   vehicle_roi_updated;
+
+static struct vehicle_global_position_s *vehicle_global_position;
+static bool   vehicle_global_position_updated;
+
+static struct vehicle_mount_s *vehicle_mount;
+static bool   vehicle_mount_updated;
+
+static struct position_setpoint_triplet_s *position_setpoint_triplet;
+static bool   position_setpoint_triplet_updated;
+
+static struct manual_control_setpoint_s *manual_control_setpoint;
+static bool   manual_control_setpoint_updated;
+
+static struct {
+	int mnt_mode;
+	int mnt_mav_sysid;
+	int mnt_mav_compid;
+	int mnt_man_control;
+	int mnt_man_roll;
+	int mnt_man_pitch;
+	int mnt_man_yaw;
+} params;
+
+static struct {
+	param_t mnt_mode;
+	param_t mnt_mav_sysid;
+	param_t mnt_mav_compid;
+	param_t mnt_man_control;
+	param_t mnt_man_roll;
+	param_t mnt_man_pitch;
+	param_t mnt_man_yaw;
+} params_handels;
+
+
+/**
+ * Print command usage information
+ */
+static void usage()
+{
+	fprintf(stderr,
+		"usage: mount start\n"
+		"       mount stop\n"
+		"       mount status\n");
+	exit(1);
+}
+
+/**
+ * The daemon thread.
+ */
+static int mount_thread_main(int argc, char *argv[])
+{
+	if (!get_params()) {
```

Comment:
Parameters need to be updated during the loop, otherwise you have to restart for each param change. Have a look how the rest of the system does it.